x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -236,69 +268,48 @@ func getInstancesPerRegion(svc ec2iface.EC2API) (instanceIDs []string, instances
 		req := svc.DescribeInstancesRequest(describeInstanceInput)
 		output, err := req.Send()
 		if err != nil {
-			logp.Error(errors.Wrap(err, "Error DescribeInstances"))
+			err = errors.Wrap(err, "Error DescribeInstances")
 			return nil, nil, err
 		}
 
 		for _, reservation := range output.Reservations {
 			for _, instance := range reservation.Instances {
-				instanceID := *instance.InstanceId
-				instanceIDs = append(instanceIDs, instanceID)
-				instancesOutputs[instanceID] = instance
+				instanceIDs = append(instanceIDs, *instance.InstanceId)
+				instancesOutputs[*instance.InstanceId] = instance
 			}
 		}
 	}
 	return
 }
 
-func convertPeriodToDuration(period string, detailedMonitoring ec2.MonitoringState) (duration string, periodInSeconds int) {
+func convertPeriodToDuration(period string) (duration string, periodInSec int, err error) {
 	// Amazon EC2 sends metrics to Amazon CloudWatch with 5-minute default frequency.
 	// If detailed monitoring is enabled, then data will be available in 1-minute period.
 	// Set starttime double the default frequency earlier than the endtime in order to make sure
 	// GetMetricDataRequest gets the latest data point for each metric.
+	duration = ""
+	periodInSec = 0
 	numberPeriod, err := strconv.Atoi(period[0 : len(period)-1])
 	if err != nil {
-		logp.Error(errors.Wrap(err, "Error converting string to int. Use default duration instead."))
-		// If failed converting string to int, then set default duration to "-600s" with basic monitoring and "-120s" with
-		// detailed monitoring.
-		numberPeriod = 300
-		if detailedMonitoring == ec2.MonitoringStateEnabled {
-			numberPeriod = 60
-		}
-		duration = "-" + strconv.Itoa(numberPeriod*2) + "s"
-		periodInSeconds = numberPeriod
-		return duration, periodInSeconds
+		return
```

Comment:
Not a big fan of name returns as it makes the code harder to read. (my personal opinion). There are cases where it is needed like in defer statements but I would not use it here to keep the code path obvious.

Reply:
Yeah sounds good, I will change it!